### PR TITLE
Add table oci_cloud_guard_managed_list. Closes #70

### LINKED ISSUE #70: Add table oci_cloud_guard_managed_list
**References**
We need ensure Cloud Guard is enabled in the root compartment of the tenancy as per CIS requirements.

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/oci_cloud_guard_managed_list []

PRETEST: tests/oci_cloud_guard_managed_list

TEST: tests/oci_cloud_guard_managed_list
Running terraform
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci cloud-guard managed-list list --compartment-id ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq --output json > /private/var/folders/d8/gsp6f_cn3359xnt_gs3cts_80000gn/T/tests/oci_cloud_guard_managed_list/terraform/test/output.json"]
null_resource.named_test_resource: Still creating... [10s elapsed]
null_resource.named_test_resource (local-exec): WARNING: This operation supports pagination and not all resources were returned.  Re-run using the --all option to auto paginate and list all resources.
null_resource.named_test_resource: Creation complete after 15s [id=3630657459676037593]
data.local_file.input: Reading...
data.local_file.input: Read complete after 0s [id=75a8dba42b7fe48145cadac5831d8af6dbffddc9]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

resource_id = "ocid1.cloudguardmanagedlist.oc1.ap-mumbai-1.amaaaaaa6igdexcqcqtpjx76nlylrevdughl5fmwcky2mivgqy5bproriumq"
resource_name = "Oracle Cloudguard CIDR Managed List"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

Running SQL query: test-get-query.sql

[
  {
    "id": "ocid1.cloudguardmanagedlist.oc1.ap-mumbai-1.amaaaaaa6igdexcqcqtpjx76nlylrevdughl5fmwcky2mivgqy5bproriumq",
    "lifecycle_state": "ACTIVE",
    "name": "Oracle Cloudguard CIDR Managed List"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "id": "ocid1.cloudguardmanagedlist.oc1.ap-mumbai-1.amaaaaaa6igdexcqcqtpjx76nlylrevdughl5fmwcky2mivgqy5bproriumq",
    "lifecycle_state": "ACTIVE",
    "name": "Oracle Cloudguard CIDR Managed List"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "Oracle Cloudguard CIDR Managed List"
  }
]
✔ PASSED

POSTTEST: tests/oci_cloud_guard_managed_list

TEARDOWN: tests/oci_cloud_guard_managed_list

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  name,
  id,
  time_created,
  lifecycle_state as state
from
  oci_cloud_guard_managed_list;
+-------------------------------------------+----------------------------------------------------------------------------------------------------------+---------------------+--------+
| name                                      | id                                                                                                       | time_created        | state  |
+-------------------------------------------+----------------------------------------------------------------------------------------------------------+---------------------+--------+
| Oracle Cloudguard CIDR Managed List       | ocid1.cloudguardmanagedlist.oc1.ap-mumbai-1.amaaaaaa6igdexcqpuuqmtzatq7st2snr6aj26uounvtzfnyd6n2qmuh7kfq | 2021-04-12 07:53:46 | ACTIVE |
| Oracle Cloudguard CIDR Managed List-Clone | ocid1.cloudguardmanagedlist.oc1.ap-mumbai-1.amaaaaaa6igdexaazuz2y3fdfhkpemx7wg6q7sqkddthpnuosud5bgv7eymq | 2021-04-13 05:20:28 | ACTIVE |
+-------------------------------------------+----------------------------------------------------------------------------------------------------------+---------------------+--------+

select
  name,
  id,
  time_created,
  lifecycle_state as state
from
  oci_cloud_guard_managed_list where lifecycle_state = 'ACTIVE'
+-------------------------------------------+----------------------------------------------------------------------------------------------------------+---------------------+--------+
| name                                      | id                                                                                                       | time_created        | state  |
+-------------------------------------------+----------------------------------------------------------------------------------------------------------+---------------------+--------+
| Oracle Cloudguard CIDR Managed List       | ocid1.cloudguardmanagedlist.oc1.ap-mumbai-1.amaaaaaa6igdexcqpuuqmtzatq7st2snr6aj26uounvtzfnyd6n2qmuh7kfq | 2021-04-12 07:53:46 | ACTIVE |
| Oracle Cloudguard CIDR Managed List-Clone | ocid1.cloudguardmanagedlist.oc1.ap-mumbai-1.amaaaaaa6igdexaazuz2y3fdfhkpemx7wg6q7sqkddthpnuosud5bgv7eymq | 2021-04-13 05:20:28 | ACTIVE |
+-------------------------------------------+----------------------------------------------------------------------------------------------------------+---------------------+--------+

```
</details>

### CODE REVIEW

**Comment by bigdatasourav** [comment]
```
SETUP: tests/oci_cloud_guard_managed_list []

PRETEST: tests/oci_cloud_guard_managed_list

TEST: tests/oci_cloud_guard_managed_list
Running terraform
oci_cloud_guard_managed_list.named_test_resource: Creating...
oci_cloud_guard_managed_list.named_test_resource: Still creating... [10s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [20s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [30s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [40s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [50s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [1m0s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [1m10s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [1m20s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [1m30s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [1m40s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [1m50s elapsed]
oci_cloud_guard_managed_list.named_test_resource: Still creating... [2m0s elapsed]

Error: 500-InternalError 
Provider version: 4.22.0, released on 2021-04-14.  
Service: Cloud Guard Managed List 
Error Message: Internal error occurred 
OPC request ID: a29de305e86e61403f7f6a68e98fcc31/7C2F5E6465C69EE7A91E1DA6C21B1023/4877569365D97B16C02E5B21EBE4C339 
Suggestion: The service for this resource encountered an error. Please contact support for help with service: Cloud Guard Managed List


  on variables.tf line 30, in resource "oci_cloud_guard_managed_list" "named_test_resource":
  30: resource "oci_cloud_guard_managed_list" "named_test_resource" {


Terraform run failed, skipping SQL queries

✘ FAILED
Terraform run failed, skipping SQL queries

✘ FAILED
Terraform run failed, skipping SQL queries

✘ FAILED
Terraform run failed, skipping SQL queries

✘ FAILED

TEARDOWN: tests/oci_cloud_guard_managed_list

SUMMARY:

✘ tests/oci_cloud_guard_managed_list failed.

0/1 passed.

**Comment by cbruno10** [comment]
Closing draft PR, will re-open once blockers are cleared.

**Comment by bigdatasourav** [comment]
test issue resolved.

File: oci/table_ oci_cloud_guard_managed_list.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeUpdated.Time"),
+			},
+			{
+				Name:        "list_type",
+				Description: "Type of the list.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "feed_provider",
```
Comment (khushboo9024) [inline oci/table_ oci_cloud_guard_managed_list.go:78]: Columns are not in alphabetical order.

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Minor changes required.

File: oci/table_ oci_cloud_guard_managed_list.go (hunk 1 above)
Comment (bigdatasourav) [reply oci/table_ oci_cloud_guard_managed_list.go]: as discussed, updated.

**Review by khushboo9024 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**